kiwi/solverimpl.h

```diff
@@ -605,7 +605,9 @@ class SolverImpl
 			Symbol leaving( m_infeasible_rows.back() );
 			m_infeasible_rows.pop_back();
 			RowMap::iterator it = m_rows.find( leaving );
-			if( it != m_rows.end() && it->second->constant() < 0.0 )
+			if( it != m_rows.end() &&
+			    !( allDummies( *it->second ) && nearZero( it->second->constant() ) ) &&
```

Comment:
A row with all dummies should be near zero, so that alone should be a sufficient check. No need to iterate the row to check for all dummies.